server/proxy/pf_client.c

```diff
@@ -222,16 +229,37 @@ static void pf_client_post_disconnect(freerdp* instance)
 static DWORD WINAPI pf_client_thread_proc(LPVOID arg)
 {
 	freerdp* instance = (freerdp*)arg;
+	pClientContext* pc = (pClientContext*)instance->context;
 	DWORD nCount;
 	DWORD status;
 	HANDLE handles[64];
 
+	pc->during_connect_process = TRUE;
 	if (!freerdp_connect(instance))
 	{
-		WLog_ERR(TAG, "connection failure");
-		return 0;
+		if (instance->settings->NlaSecurity)
+		{
+			WLog_ERR(TAG, "freerdp_connect() failed, trying to connect without NLA");
+			/* disable NLA, enable TLS */
+			instance->settings->NlaSecurity = FALSE;
+			instance->settings->TlsSecurity = TRUE;
```

Comment:
Should we explicitly enable RdpSecurity too?

Reply:
it is already set to true, but for the code to be more readable, yes i'll add it